### PR TITLE
Vic3 miscellaneous validations

### PR DESCRIPTION
**Pull request by amtep**

### CODE REVIEW

File: src/vic3/data/governments.rs [hunk 1]
```diff
@@ -49,3 +50,32 @@ impl DbKind for GovernmentType {
...
+        db.add(Item::LegitimacyLevel, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for LegitimacyLevel {
+    fn validate(&self, key: &Token, block: &Block, data: &Everything) {
+        // This entire item type is undocumented
+        let mut vd = Validator::new(block, data);
+
+        data.verify_exists(Item::Localization, key);
```
Comment (my4ng) [inline src/vic3/data/governments.rs:72]: also legitimacy_level_*_loyalties_gain loca

File: src/vic3/data/map.rs [hunk 2]
```diff
@@ -45,3 +45,134 @@ impl DbKind for MapLayer {
...
 }
+
+#[derive(Clone, Debug)]
+pub struct MapMode {}
+
+inventory::submit! {
+    ItemLoader::Normal(GameFlags::Vic3, Item::MapMode, MapMode::add)
+}
+
+// LAST UPDATED VIC3 VERSION 1.5.13
```
Comment (my4ng) [inline src/vic3/data/map.rs:56]: Best mark where these are from: `gfx/map/map_modes/map_modes.txt`

File: src/vic3/data/map.rs [hunk 3]
```diff
@@ -45,3 +45,134 @@ impl DbKind for MapLayer {
...
+    &["default", "heatmap", "interactive", "mapmode_extra", "states"];
+
+impl MapMode {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::MapMode, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for MapMode {
+    fn validate(&self, _key: &Token, block: &Block, data: &Everything) {
```
Comment (my4ng) [inline src/vic3/data/map.rs:111]: verify loca too

**Review by my4ng — CHANGES_REQUESTED** [review]
All good except minor issues

File: src/vic3/data/map.rs (hunk 2 above)
Comment (amtep) [reply src/vic3/data/map.rs:56]: Turns out that gradient border settings have a .txt file defining them :) I'll add a loader for that too

File: src/vic3/data/map.rs (hunk 3 above)
Comment (amtep) [reply src/vic3/data/map.rs:111]: Done

**Review by my4ng — APPROVED** [review]

**Pull request merged**